contract/src/airdrop.contract.js

```diff
@@ -367,21 +424,54 @@ export const start = async (zcf, privateArgs, baggage) => {
             );
 
             const paymentAmount = this.state.payoutArray[tier];
+            console.log('------------------------');
+            console.log('paymentAmount::', paymentAmount);
+            const payment = await withdrawFromSeat(zcf, tokenHolderSeat, {
+              Tokens: paymentAmount,
+            });
+
+            console.log('------------------------');
+            console.log(
+              'withdrawn from tokenHolderSeat ### payment::',
+              payment,
+            );
+            console.log('------------------------');
+
+            const depositFacet = await getDepositFacet(derivedAddress);
+
+            console.log('------------------------');
+            console.log(
+              `depositFacet:: AFTER getDepositFacet(${derivedAddress})`,
+              depositFacet,
+            );
+            await Promise.all(
+              Object.values(payment).map(pmtP =>
+                E.when(pmtP, pmt => E(depositFacet).receive(pmt)),
+              ),
+            );
+            // XXX partial failure? return payments?
+            // await Promise.all(
+            //   E.when(
+            //     payment,
+            //     E(depositFacet).receive(payment),
+            //     new Error('Error getting payment'),
+            //   ),
+            // );
+            TT(
+              'After await E.when(payment, pmy => E(depositFacet).recieve(pmt)',
+            );
 
             rearrange(
-              harden([
-                [tokenHolderSeat, claimSeat, { Tokens: paymentAmount }],
-                [claimSeat, tokenHolderSeat, { Fee: claimTokensFee }],
-              ]),
+              harden([[claimSeat, tokenHolderSeat, { Fee: claimTokensFee }]]),
             );
```

Comment:
We've opted to allow for the offer to occur in such a manner enforces offer-safety for the `tokenHolderSeat`. This is due to the nature of this specific offer as it uses `depositFacet`s one one side of the offer, and Zoe seat rearrange for the other. 

If this is an issue, then I think the fix would be to move the accountStore logic so that it occurs before the `depositFacet` is sent a payment, effectively stopping any instances in which a user manages to get passed the check on `accountStore` before the  `Promise.all` payment resolves. 

Does that make sense?

Reply:
I've realized that a merge I conducted for adding [this commit](https://github.com/hindley-milner-systems/dapp-ertp-airdrop/pull/151/commits/51d3b7969e1015813144891007de3d94b4f0b18e) into the codebase got lost along the way. I'm going to address this today. I wanted to make note of it though as it relates to ensuring offer safety.